paimon-core/src/main/java/org/apache/paimon/table/AppendOnlyFileStoreTable.java

```diff
@@ -91,13 +90,9 @@ protected SplitGenerator splitGenerator() {
                 bucketMode());
     }
 
-    /**
-     * Currently, the streaming read of overwrite is implemented by reversing the {@link RowKind} of
-     * overwrote records to {@link RowKind#DELETE}, so only tables that have primary key support it.
-     */
     @Override
     public boolean supportStreamingReadOverwrite() {
-        return false;
+        return new CoreOptions(tableSchema.options()).streamingReadAppendOverwrite();
```

Comment:
Before this modify, the code note says "Currently, the streaming read of overwrite is implemented by reversing the {@link RowKind} of overwrote records to {@link RowKind#DELETE}, so only tables that have primary key support it."

Way this pr can support streaming read overwrite snapshot of append table directly?

Reply:
Yes, this pr introduce `streaming-read-append-overwrite`, when enabled, the delta parts of the `OVERWRITE` commit will be read.